mm2src/lp_ordermatch.rs

```diff
@@ -1990,13 +1987,12 @@ fn order_pair_root_mut<'a>(state: &'a mut HashMap<AlbOrderedOrderbookPair, H64>,
 }
 
 fn pair_history_mut<'a>(
-    state: &'a mut HashMap<AlbOrderedOrderbookPair, TrieOrderHistory>,
+    state: &'a mut TimeCache<AlbOrderedOrderbookPair, TrieOrderHistory>,
     pair: &str,
 ) -> &'a mut TrieOrderHistory {
-    match state.raw_entry_mut().from_key(pair) {
-        RawEntryMut::Occupied(e) => e.into_mut(),
-        RawEntryMut::Vacant(e) => e.insert(pair.to_owned(), Default::default()).1,
-    }
+    state.entry(pair.into()).or_insert_with(|| TrieOrderHistory {
```

Comment:
As per, https://github.com/KomodoPlatform/atomicDEX-API/pull/1054#discussion_r700984546, `or_insert_with_update_expiration` is supposed to be used here.